src/python/director/drakevisualizer.py

```diff
@@ -215,62 +240,68 @@ def loadPolyDataMeshes(geom):
             print 'warning, cannot find file:', filename
             return []
 
+        mesh_vis_list = []
+
         if filename.endswith('vtm'):
             polyDataList = ioUtils.readMultiBlock(filename)
+            mesh_vis_list = Geometry.DeriveMeshVisList(geom = geom, \
+                meshFileName = filename, polyDataList = polyDataList)
         else:
-            polyDataList = [ioUtils.readPolyData(filename)]
-
-        if USE_TEXTURE_MESHES:
-            for polyData in polyDataList:
-                Geometry.loadTextureForMesh(polyData, filename)
-
-
-        Geometry.MeshCache[filename] = list(polyDataList)
-
-        return polyDataList
+            if os.path.isfile(basename + '.mtl'):
```

Comment:
bug: this is not the correct way to determine whether an mtl file exists.  Also, we probably want all obj files to be loaded with the `ioUtils.readObjMtl` function, not just obj files with `.mtl` material files.